platform/darwin/src/NSCoder+MGLAdditions.h

```diff
@@ -0,0 +1,12 @@
+#import <Foundation/Foundation.h>
+#import <CoreLocation/CoreLocation.h>
+
+#import <mbgl/util/feature.hpp>
+
+@interface NSCoder (MGLAdditions)
+
+- (void)mgl_encodeLocationCoordinates2D:(std::vector<CLLocationCoordinate2D>)coordinates forKey:(NSString *)key;
```

Comment:
I think it would be useful to publicly expose a method that encodes a single CLLocationCoordinate2D. It would be called `-encodeMGLCoordinate:`, for consistency with existing category methods on NSCoder. (See also `-[NSValue valueWithMGLCoordinate:]`.)